examples/posenet/PoseNet.js

```diff
@@ -0,0 +1,361 @@
+    }
+
+    _calculateoutput(input_dimension, shape, stride, layer){
+        var padding = 1;
+        var output_dimension;
+        if(layer == "conv2d"){
+            output_dimension = [1, Math.floor((input_dimension[1]-shape[1]+2)/stride+1), Math.floor((input_dimension[2]-shape[2]+2)/stride+1), shape[0]];
+        }
+        else if(layer == "depthwise"){
+            output_dimension = [1, Math.floor((input_dimension[1]-shape[1]+2)/stride+1), Math.floor((input_dimension[2]-shape[2]+2)/stride+1), input_dimension[3]];
+        }
+        else{
+            output_dimension = [1, Math.floor((input_dimension[1]-shape[1]+0)/stride+1), Math.floor((input_dimension[2]-shape[2]+0)/stride+1), shape[0]];
+        }
+        
+        //console.log(output_dimension);
+        return output_dimension;
+    }
+
+
+    _addScalarInt32(value){
+        const scalarInt32Type = {type: this._nn.INT32};
+        let index = this._operandIndex++;
+        this._model.addOperand(scalarInt32Type);
+        this._model.setOperandValue(index, new Int32Array([value]));
+        return index;
+    }
+    _addScalarFloat32(value) {
+        const scalarInt32Type = {type: this._nn.FLOAT32};
+        let index = this._operandIndex++;
+        this._model.addOperand(scalarInt32Type);
+        this._model.setOperandValue(index, new Float32Array([value]));
+        return index;
+  }
+    
+    _addOpsAndParams(){
+        let index = 0;
+        for(let i in this._tfmodel){
+            if(this._tfmodel[i].convType=="conv2d"){
+                let opCode = 3; //conv
+                let inputs = [];
+                inputs.push(this._model._inputs[0]);
+                let outputs = this._outputs[index];
+                let paddingCode = 1;
+                inputs.push(outputs+1);
+                inputs.push(outputs+2);
+                inputs.push(this._addScalarInt32(paddingCode));
+                inputs.push(this._addScalarInt32(this._tfmodel[i].stride));
+                inputs.push(this._addScalarInt32(this._tfmodel[i].stride));
+                let fuseCode = 3; //relu6 activition function
+                inputs.push(this._addScalarInt32(fuseCode));
+                let opType = this._nn.CONV_2D;
+                this._model.addOperation(opType, inputs, [outputs]);
+            }
+            else if(this._tfmodel[i].convType == "separableConv"){
+                //depthwise conv
+                let inputs = [];
+                inputs.push(this._outputs[index]);
+                index++;
+                let outputs = this._outputs[index];
+                let paddingCode = 1; //SAME
+                inputs.push(outputs+2);
+                inputs.push(outputs+3);
+                inputs.push(this._addScalarInt32(paddingCode));
+                inputs.push(this._addScalarInt32(this._tfmodel[i].stride));
+                inputs.push(this._addScalarInt32(this._tfmodel[i].stride));
+                if(this._tfmodel[i].rate !==1){
```

Comment:
Please comment the workaround to reuse `multiplier` for `dilation rate`. And where is your modification to WebGL2 backend for dilation workaround?